test/services/reply-comments.test.js

```diff
@@ -0,0 +1,102 @@
+const assert = require('assert');
+const app = require('../../server/app');
+const service = app.service('comments');
+const contributionService = app.service('contributions');
+const userService = app.service('users');
+
+const { userData } = require('../assets/users');
+const { contributionData } = require('../assets/contributions');
+const { commentData } = require('../assets/comments');
+
+const replyData = {
+  content: 'Nested comment'
+};
+
+const replyUserData = {
+  email: 'test3@test3.de',
+  password: '1234',
+  name: 'Jack',
+  timezone: 'Europe/Berlin',
+  badgeIds: [],
+  role: 'user'
+};
+
+
+describe('\'reply comments\' service', () => {
+  let user;
+  let replyUser;
+  let params, replyParams;
+  let contribution;
+  let comment;
+
+  before(function(done) {
+    this.server = app.listen(3031);
+    this.server.once('listening', () => done());
+  });
+
+  after(function(done) {
+    this.server.close(done);
+  });
+
+  beforeEach(async () => {
+    await app.get('mongooseClient').connection.dropDatabase();
+    user = await userService.create(userData);
+    replyUser = await userService.create(replyUserData);
+
+    params = {
+      user
+    };
+    replyParams = {
+      replyUser
+    };
+
+    contribution = await contributionService.create(contributionData, params);
+
+    commentData.userId = user._id;
+    commentData.contributionId = contribution._id;
+
+    replyData.userId = replyUser._id;
+    replyData.contributionId = contribution._id;
+  });
+
+  afterEach(async () => {
+    await app.get('mongooseClient').connection.dropDatabase();
+    user = null;
+    params = null;
+    contribution = null;
+    comment = null;
+    delete commentData.userId;
+    delete commentData.contributionId;
+    delete replyData.userId;
+    delete replyData.contributionId;
```

Comment:
I think you don't need to reset the shared data here. Probably you are afraid of some tests interfering with other tests because of stale data? There are different approaches here, but I will share mine:

I make use of nested scenarios. If more than one test needs some data, I write `context('given a reply user', () => {`, put the `let` statement within the block and setup the user in a `beforeEach` block. I don't reset the data. You will quickly forget to reset all your data anyways. But within the context block, all tests should be OK with the preexisting data. If some tests do not require the data to be setup, they don't belong into that block.

Do you see what I mean?

Reply:
Ok I can see the point. It's also good for the test performance if the data is not created per test, but per context.